gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineModuleJvmArgs.java

```diff
@@ -92,6 +96,62 @@ public void apply(Project project) {
                                 return arguments;
                             }
                         });
+
+                Task maybeJavadocTask = project.getTasks().findByName(sourceSet.getJavadocTaskName());
+                if (maybeJavadocTask instanceof Javadoc) {
+                    maybeJavadocTask.doFirst(new Action<Task>() {
+                        @Override
+                        public void execute(Task task) {
+                            Javadoc javadoc = (Javadoc) task;
+
+                            MinimalJavadocOptions options = javadoc.getOptions();
+                            if (options instanceof CoreJavadocOptions) {
+                                CoreJavadocOptions coreOptions = (CoreJavadocOptions) options;
+                                ImmutableList<JarManifestModuleInfo> info = collectClasspathInfo(project, sourceSet);
+                                List<String> exportValues = Stream.concat(
+                                                extension.exports().get().stream(),
+                                                info.stream().flatMap(item -> item.exports().stream()))
+                                        .distinct()
+                                        .sorted()
+                                        .map(item -> item + "=ALL-UNNAMED")
+                                        .collect(ImmutableList.toImmutableList());
+                                List<String> opens = Stream.concat(
+                                                extension.opens().get().stream(),
+                                                info.stream().flatMap(item -> item.opens().stream()))
+                                        .distinct()
+                                        .sorted()
+                                        .map(item -> item + "=ALL-UNNAMED")
+                                        .collect(ImmutableList.toImmutableList());
+                                project.getLogger()
+                                        .debug(
+                                                "BaselineModuleJvmArgs building {} on {} "
+                                                        + "with exports: {} and opens: {}",
+                                                javadoc.getName(),
+                                                project,
+                                                exportValues,
+                                                opens);
+                                if (!exportValues.isEmpty()) {
+                                    coreOptions
+                                            // options are automatically prefixed with '-' internally
+                                            .addMultilineStringsOption("-add-exports")
```

Comment:
Was wondering why we can't use `collectClasspathArgs` here which does a very similar colelction of the export and open args. Is it because of the `--` vs `-` distinction?
https://github.com/palantir/gradle-baseline/blob/159bb3811ffb9aa053e10a202d43c0e7827cb8e1/gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineModuleJvmArgs.java#L261

Reply:
Sort of, `add-exports` only occurs once, and the values are collected on the right-hand-side.